components/formats-bsd/test/loci/formats/utests/MemoizerTest.java

```diff
@@ -32,315 +32,201 @@
-
   private String id;
-
   private FakeReader reader;
 
-  private Memoizer memoizer;
+  private static File createTempDir() throws Exception {
+    return Files.createTempDirectory(TMP_PREFIX).toFile();
+  }
 
-  private static int fullPlaneCallIndex;
+  private static void recursiveDeleteOnExit(File rootDir) {
+    rootDir.deleteOnExit();
+    File[] children = rootDir.listFiles();
+    if (null != children) {
+      for (File child: children) {
+        if (child.isDirectory()) {
+          recursiveDeleteOnExit(child);
+        } else {
+          child.deleteOnExit();
+        }
+      }
+    }
+  }
 
-  private int sizeX;
+  private static void checkMemo(Memoizer memoizer, String id)
+      throws Exception {
+    memoizer.setId(id);
+    assertFalse(memoizer.isLoadedFromMemo());
+    assertTrue(memoizer.isSavedToMemo());
+    memoizer.close();
+    memoizer.setId(id);
+    assertTrue(memoizer.isLoadedFromMemo());
+    assertFalse(memoizer.isSavedToMemo());
+    memoizer.close();
+  }
 
-  private int sizeY;
+  private static void checkNoMemo(Memoizer memoizer, String id)
+      throws Exception {
+    memoizer.setId(id);
+    assertFalse(memoizer.isLoadedFromMemo());
+    assertFalse(memoizer.isSavedToMemo());
+    memoizer.close();
+  }
 
-  private int bpp;
+  private void checkMemoFile(File memoFile) {
+    checkMemoFile(memoFile, idDir);
+  }
 
-  private int planeSize;
+  private void checkMemoFile(File memoFile, File memoDir) {
+    File expMemoFile = new File(memoDir, "." + TEST_FILE + ".bfmemo");
+    assertEquals(memoFile.getAbsolutePath(), expMemoFile.getAbsolutePath());
+  }
 
   @BeforeMethod
   public void setUp() throws Exception {
-    fullPlaneCallIndex = 1;
-    // No mapping.
-    // Location.mapId(TEST_FILE, TEST_FILE);
-    reader = new FakeReader();
-    try {
-      String uuid = UUID.randomUUID().toString();
-      idDir = new File(System.getProperty("java.io.tmpdir"), uuid);
-      idDir.mkdirs();
-      File tempFile = new File(idDir, TEST_FILE);
-      tempFile.createNewFile();
-      id = tempFile.getAbsolutePath();
-      reader.setId(id);
-      sizeX = reader.getSizeX();
-      sizeY = reader.getSizeY();
-      bpp = FormatTools.getBytesPerPixel(reader.getPixelType());
-      planeSize = sizeY * sizeY * bpp;
-    } finally {
-      reader.close();
-    }
+    idDir = createTempDir();
+    File tempFile = new File(idDir, TEST_FILE);
+    tempFile.createNewFile();
+    id = tempFile.getAbsolutePath();
     reader = new FakeReader(); // No setId !
   }
 
   @AfterMethod
   public void tearDown() throws Exception {
-    memoizer.close();
     reader.close();
-  }
-
-  @Test
-  public void testSimple() throws Exception {
-      memoizer = new Memoizer(reader);
-      File f = memoizer.getMemoFile(id);
-      if (f != null && f.exists()) {
-        f.delete();
-      }
-      // At this point we're sure that there's no memo file.
-      reader.setId(id);
-      reader.close();
-      memoizer.setId(id);
```

Comment:
The reason for removing that test was that it wasn't performing any explicit checks. I've just added a new one that performs the open/close on the wrapped reader, see if it looks useful.